docs/docs/the-component-lifecycle.md

```diff
@@ -0,0 +1,43 @@
+---
+id: the-component-lifecycle
+title: The Component Lifecycle
+permalink: docs/the-component-lifecycle.html
+prev: rendering-elements.html
+---
+
+Components have three main parts of their lifecycle:
+
+* **[Mounting](#mounting):** A component is being inserted into the DOM.
+* **[Updating](#updating):** A component is being re-rendered to determine if the DOM should be updated.
+* **[Unmounting](#unmounting):** A component is being removed from the DOM.
+
+React provides lifecycle methods that you can specify to hook into this process. Methods prefixed with **`will`** are called right before something happens, and methods prefixed with **`did`** are called right after something happens.
+
+## Mounting
+
+These methods are called when a component is being inserted into the DOM.
+
+* **`getInitialState(): object`** is invoked before a component is mounted. Stateful components should implement this and return the initial state data.
```

Comment:
This is part of the legacy API. It is only relevant for `createClass()` which we’re dedicating a separate page to. Can we somehow mark it as legacy here?

Class components use `this.state` assignment in `constructor` instead. See https://facebook.github.io/react/docs/reusable-components.html#setting-the-initial-state